cuda_core/tests/test_module.py

```diff
@@ -180,6 +180,242 @@ def test_object_code_handle(get_saxpy_object_code):
     assert mod.handle is not None
 
 
+@pytest.fixture(scope="function")
+def get_ltoir_object_code(init_cuda):
+    # Create LTOIR code using link-time optimization
+    prog = Program(SAXPY_KERNEL, code_type="c++", options=ProgramOptions(link_time_optimization=True))
+    mod = prog.compile("ltoir", name_expressions=("saxpy<float>", "saxpy<double>"))
+    return mod
+
+
+def test_object_code_load_ltoir(get_ltoir_object_code):
+    mod = get_ltoir_object_code
+    ltoir = mod._module
+    sym_map = mod._sym_map
+    assert isinstance(ltoir, bytes)
+    mod_obj = ObjectCode.from_ltoir(ltoir, symbol_mapping=sym_map)
+    assert mod_obj.code == ltoir
+    assert mod_obj._code_type == "ltoir"
+    # ltoir doesn't support kernel retrieval directly as it's used for linking
+    assert mod_obj._handle is None  # Should only be loaded when needed
+    # Test that get_kernel fails for unsupported code type
+    with pytest.raises(RuntimeError, match=r'Unsupported code type "ltoir"'):
+        mod_obj.get_kernel("saxpy<float>")
+
+
+def test_object_code_load_ltoir_from_file(get_ltoir_object_code, tmp_path):
+    mod = get_ltoir_object_code
+    ltoir = mod._module
+    sym_map = mod._sym_map
+    assert isinstance(ltoir, bytes)
+    ltoir_file = tmp_path / "test.ltoir"
+    ltoir_file.write_bytes(ltoir)
+    mod_obj = ObjectCode.from_ltoir(str(ltoir_file), symbol_mapping=sym_map)
+    assert mod_obj.code == str(ltoir_file)
+    assert mod_obj._code_type == "ltoir"
+    assert mod_obj._handle is None  # Should only be loaded when needed
+
+
+def test_object_code_load_fatbin(get_saxpy_kernel):
```

Comment:
For the fatbin and object (library) tests, let us approach it this way
- we detect if NVCC exists in PATH, if not we skip the test
- we adopt and vendor the tests from numba-cuda (https://github.com/NVIDIA/numba-cuda/tree/0f6ea695f2a26f0e905ed91a3d2d5954cdcca42b/numba_cuda/numba/cuda/tests/test_binary_generation) to generate fatbins
   - numba-cuda should NOT be a test dependency! 
- we make up two or three archs, including `sm_75`, `sm_90`, and the current device's arch (check `Device().arch`)
- for the file-based test, we load the NVCC-generated fatbin from a temp path
- for the in-memory-based test, we preload the fatbin from disk to memory and pass the bytes
- add a TODO comment noting that we can drop NVCC from the test dependency once #156 is resolved